.github/workflows/publish_future.yml

```diff
@@ -7,6 +7,9 @@ on:
 
 jobs:
   deployment:
+    strategy:
+      matrix:
+        base_image: ["alpine:3.17", "gcr.io/google.com/cloudsdktool/google-cloud-cli:alpine"]
```

Comment:
In this situation, I'm not sure the matrix really adds much because you're still duplicating the push actions and just using conditionals to execute them. I think it'd actually be simpler to just have two steps with the base image hard-coded rather than using conditionals like this.

Reply:
good point. I guess the only advantage is parallelization but it really doesn't matter here. Refactoring it.